### PR TITLE
(feat) Update billing to new workflow on check in

### PR DESCRIPTION
**Pull request by donaldkibet**
## Requirements

- [ ] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [ ] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [ ] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary

This PR updates the billing workflow by having two categories. 
Exempted and non-exempted patient

## Screenshots
Peek 2024-02-20 03-29

## Related Issue

*None.*

## Other

*None.*

### CODE REVIEW

**Review by ojwanganto — CHANGES_REQUESTED** [review]
@donaldkibet please check if my comments hold water. Thanks

File: packages/esm-billing-app/src/billing-form/visit-attributes/visit-attributes-form.component.tsx [hunk 1]
```diff
@@ -102,88 +102,96 @@ const VisitAttributesForm: React.FC<VisitAttributesFormProps> = ({ setAttributes
...
+          <Layer className={styles.sectionFieldLayer}>
+            <Controller
+              control={control}
+              name="patientCategory"
+              render={({ field }) => (
+                <ComboBox
+                  className={styles.sectionField}
+                  onChange={({ selectedItem }) => field.onChange(selectedItem?.uuid)}
+                  id="patientCategory"
+                  items={[{ uuid: '162277AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA', display: 'In Prision' }]}
```
[the remark this answers is not in the document]
Comment (jecihjoy) [reply packages/esm-billing-app/src/billing-form/visit-attributes/visit-attributes-form.component.tsx:133]: @ojwanganto the plan here was to change from using concepts to just free text but managed using a drop down

[the remark this answers is not in the document]
Comment (ojwanganto) [reply packages/esm-billing-app/src/billing-form/visit-attributes/visit-attributes-form.component.tsx:133]: ok

**Comment by ojwanganto** [comment]
Are we good to go with this one? @donaldkibet @makombe @jecihjoy

File: packages/esm-billing-app/src/billable-services/bill-waiver/utils.ts [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
+  paymentModes: Array<OpenmrsResource>,
 ) => {
   const { cashier } = bill;
 
   const billPayment = {
     amount: parseFloat(totalAmount.toFixed(2)),
     amountTendered: parseFloat(Number(amountWaived).toFixed(2)),
     attributes: [],
-    instanceType: waiverUuid,
+    instanceType: paymentModes?.find((mode) => mode.name.toLowerCase().includes('waiver'))?.uuid,
```
Comment (donaldkibet) [inline packages/esm-billing-app/src/billable-services/bill-waiver/utils.ts:18]: Attempt to remove hard coding waiver. The waiver payment name needs to be named appropriately from the backend

**Comment by ojwanganto** [comment]
Thank you. I think we can go ahead and merge this in.

File: packages/esm-billing-app/src/billable-services/bill-waiver/utils.ts (hunk 2 above)
Comment (ojwanganto) [reply packages/esm-billing-app/src/billable-services/bill-waiver/utils.ts:18]: This should work. The backend module should ensure this is not changed. We are good with this

**Review by jecihjoy — APPROVED** [review]

**Pull request merged**